library/src/main/java/com/anjlab/android/iab/v3/BillingProcessor.java

```diff
@@ -49,17 +51,17 @@
 	 */
 	public interface IBillingHandler
 	{
-		void onProductPurchased(String productId, TransactionDetails details);
+		void onProductPurchased(String productId, @Nullable TransactionDetails details);
 
 		void onPurchaseHistoryRestored();
 
-		void onBillingError(int errorCode, Throwable error);
+		void onBillingError(int errorCode, @Nullable Throwable error);
 
 		void onBillingInitialized();
 	}
 
-	private static final Date DATE_MERCHANT_LIMIT_1 = new Date(2012, 12, 5); //5th December 2012
-	private static final Date DATE_MERCHANT_LIMIT_2 = new Date(2015, 7, 20); //21st July 2015
+	private static final Date DATE_MERCHANT_LIMIT_1 = new Date(1354694400000L); //5th December 2012
+	private static final Date DATE_MERCHANT_LIMIT_2 = new Date(1437375600000L); //21st July 2015
```

Comment:
@serggl see latest commit. I'm 99% sure that that is how calendars are used, so it should be good. The date will likely vary based on timezones, but it shouldn't be off by more than a day.